plugins/greatexpectations/flytekitplugins/greatexpectations/schema.py

```diff
@@ -0,0 +1,294 @@
+import datetime
+import logging
+import os
+import typing
+from dataclasses import dataclass
+from typing import Type
+
+import great_expectations as ge
+from dataclasses_json import dataclass_json
+from great_expectations.checkpoint import SimpleCheckpoint
+from great_expectations.core.run_identifier import RunIdentifier
+from great_expectations.core.util import convert_to_json_serializable
+from great_expectations.exceptions import ValidationError
+
+from flytekit import FlyteContext
+from flytekit.extend import TypeEngine, TypeTransformer
+from flytekit.models import types as _type_models
+from flytekit.models.literals import Literal, Primitive, Scalar
+from flytekit.models.types import LiteralType
+from flytekit.types.file.file import FlyteFile, FlyteFilePathTransformer
+from flytekit.types.schema.types import FlyteSchema, FlyteSchemaTransformer, SchemaOpenMode
+
+
+@dataclass_json
+@dataclass
+class BatchConfig(object):
+    """
+    Use this configuration to configure Batch Request. A BatchRequest can either be
+    a simple BatchRequest or a RuntimeBatchRequest.
+
+    Args:
+        data_connector_query: query to request data batch
+        runtime_parameters: parameters to be passed at runtime
+        batch_identifiers: identifiers to identify the data batch
+        batch_spec_passthrough: reader method if your file doesn't have an extension
+        limit: number of data points to fetch
+    """
+
+    data_connector_query: typing.Optional[typing.Dict[str, typing.Any]] = None
+    runtime_parameters: typing.Optional[typing.Dict[str, str]] = None
+    batch_identifiers: typing.Optional[typing.Dict[str, str]] = None
+    batch_spec_passthrough: typing.Optional[typing.Dict[str, typing.Any]] = None
+    limit: typing.Optional[int] = None
+
+
+@dataclass_json
+@dataclass
+class GEConfig(object):
+    """
+    Use this configuration to configure GE Plugin.
+
+    Args:
+        data_source: tell where your data lives and how to get it
+        expectation_suite: suite which consists of the data expectations
+        data_connector: connector to identify data batches
+        local_file_path: dataset file path useful for FlyteFile and FlyteSchema
+        checkpoint_params: optional SimpleCheckpoint parameters
+        batchrequest_config: batchrequest config
+        data_context: directory in which GE's configuration resides
+    """
+
+    data_source: str
+    expectation_suite: str
+    data_connector: str
+    """
+    local_file_path is a must in two scenrios:
+    * When using FlyteSchema
+    * When using FlyteFile for remote paths
+    This is because base directory which has the dataset file 'must' be given in GE's config file
+    """
+    local_file_path: str = None
+    checkpoint_params: typing.Optional[typing.Dict[str, typing.Union[str, typing.List[str]]]] = None
+    batchrequest_config: BatchConfig = None
```

Comment:
Should we rename this to `batch_config` to match the type?

Reply:
It's now `BatchRequestConfig`.

Reply:
Thanks so much for all the changes, @samhita-alla - looks great ! We'll review the Great Expectations example later today or Monday of next week.

Reply:
Awesome!